compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
+-- below (which relies on the behavior of `package_app` deploying to
+-- '~/.daml/sdk/x.x.x/damlc/resources/daml-ide-core/data') selects one
+-- or the other depending on context.
+getHlintDataDir :: IO FilePath
+getHlintDataDir = do
+  root <- locateRunfiles $
+    mainWorkspace </> "compiler/damlc/daml-ide-core"
+  let test = root </> "data/test"
+      prod = root </> "daml-ide-core/data/prod"
+  useProd <- System.Directory.doesDirectoryExist prod
```

Comment:
Not particularly fond of this logic. Presumably the reason for why you want different config files is that we do not care about hlint in our tests? If so, I would suggest that we just add an explicit option to disable hlint completely in `DA.Daml.Options.Types` instead of handling it implicitly based on whether a file is present or not. This would also allow us to switch it on and off for individual tests without having to make a completely separate test with different runfiles.

Reply:
I really like the idea of being able to turn it on or off. It's not the whole story though. There's still the issue of wanting to enable it but having a different "global" config in testing vs. the IDE.

Reply:
Couldn’t we just have only a 1 global config for now and once we add support for allowing users to change the config, we use that to also specify a different config in tests?

Reply:
This is basically what we have now. A "poor mans way" of allowing users to change the config -  the directory given to hlint (which contains a hlint.yaml) file.